source/Optimizers/nelder_mead.d

```diff
@@ -0,0 +1,513 @@
+module source.optimizers.nelder_mead;
+
+import std.algorithm;
+import std.conv: to;
+import std.functional: toDelegate;
+import std.math: abs, pow, sqrt, cos;
+import std.range: enumerate;
+import std.stdio: writeln, write;
+import std.typecons: Yes;
+
+import source.Matrix;
+import source.NeuralNetwork;
+import source.Parameter;
+
+
+T nelder_mead(T)(ref Vector!T _v, T function(in Vector!T) _obj,
+                 T reflection_coef=1.0,
+                 T expension_coef=2.0,
+                 T contraction_coef=0.5,
+                 T shrink_coef=0.5) {
+    return nelder_mead!T(_v, toDelegate(_obj),
+                         reflection_coef, expension_coef,
+                         contraction_coef, shrink_coef);
+}
+
+T nelder_mead(T)(ref Vector!T _v, T delegate(in Vector!T) _obj,
+                 in T _random_bound=5.0,

```

Comment:
If you are talking about the coef, they came from the intial Paper and are the same in the sklearn implementation. However for the random bounds and maxIterations you're right, their implementation is interesting (for the maxter)
But I don't really like the implementation of the simplex. they seems to take a point, and create a simplex by modifying one axis per point on the simplex.

And they have no way to scale the simplex so _random_bound don't really have an equivalent parameter in their implementation. Also, in our implementation, we center the simplex around the origin. I made this choice because it seems a reasonable expectation in the special case of Neural Networks no ?